types/vfile/index.d.ts

```diff
@@ -0,0 +1,72 @@
+// Type definitions for vfile 2.2
+// Project: https://github.com/vfile/vfile#readme
+// Definitions by: Junyoung Choi <https://github.com/rokt33r>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare namespace vfile {
+    type Contents = string | Buffer;
+
+    interface Point {
+        line: number;
+        column: number;
+    }
+
+    interface Position {
+        start: Point;
+        end: Point;
+    }
+
+    interface VFileParams {
+        contents?: Contents;
+        path?: string;
+        basename?: string;
+        stem?: string;
+        extname?: string;
+        dirname?: string;
+        cwd?: string;
+    }
+
+    interface VFileError {
+        ruleId: string | null;
+        name: string;
+        file: string;
+        reason: string;
+        line: number | null;
+        column: number | null;
+        location: Position;
+        source: string | null;
+        fatal?: boolean | null;
+    }
+
+    type Message = (reason: string, position?: Point | Position, ruleId?: string) => VFileError;
+
+    type Fail = (reason?: string, position?: Point | Position, ruleId?: string) => void;
+
+    type Info = (reason?: string, position?: Point | Position, ruleId?: string) => void;
```

Comment:
@wooorm I've also found, position can be `Point` or `Position`. Do I understand right?

Reply:
Yes! Or a [`Node`](https://github.com/syntax-tree/unist#node)!

Reply:
Fixed!